test/blackbox-tests/test-cases/pkg/gh10959.t

```diff
@@ -34,6 +34,11 @@ Repro `dune exec --watch` crash with pkg management
   >  (system "echo building test"))
   > EOF
 
-  $ dune exec -w ./x.exe 2>&1 | grep -io "I must not crash"
-  I must not crash
-
+  $ dune exec -w ./x.exe > output.log 2>&1 &
+  $ PID=$!
+  $ sleep 2
```

Comment:
`2` seems long, what about `0.1`? The other `exec-watch` tests appear to pick something like that. Although killing processes this was seems very fragile, there ought to be a more principled way to test these. Not this PRs problem however.

Reply:
The watching/ tests use some better helpers. Up to you if you want to improve things. Also I would move this from `pkg` to `exec-watch`.

Reply:
They have these `start_dune` `stop_dune` helpers from a shared script (see the dune file for where it is). A test example is: https://github.com/ocaml/dune/blob/main/test/blackbox-tests/test-cases/watching/copy-rules.t

The logic in the script handles stopping the watch mode build and the output. Presumably something like that could be adapted for `exec`. But if this is too much work that is also fine, since the other `exec-watch/` tests are written this way.

Reply:
so this actually takes a bit longer so I added a loop that sleeps for 0.1 seconds (with a timeout) until we get the expected number of lines in the output file